src/components/activity-edit-form/state/useActivityForm.ts

```diff
@@ -15,7 +15,7 @@ import { ActionType, ActivityFormAlerts } from './activityForm.enums';
 import { copyDate } from '../../../utils/time';
 
 const defaultState: DefaultState = {
-    iconKey: getRandomIconKey(),
+    iconKey: () => getRandomIconKey(),
```

Comment:
Instead of allocating an arrow function, you should directly reference the function like so:
```suggestion
    iconKey: getRandomIconKey,
```